src/v/cloud_storage/partition_recovery_manager.h

```diff
@@ -132,6 +132,9 @@ class partition_downloader {
     struct offset_range {
         model::offset min_offset;
         model::offset max_offset;
+
+        friend auto operator<=>(const offset_range&, const offset_range&)
```

Comment:
why do we need `friend` declaration?

Reply:
I think `friend` is required here not because of data visibility, but because it's the syntax/convention? for defaulting the implementation as a free function (rather than the `auto operator<=>(const offset_range&) = default` case of a member method)